invisible_cities/reco/paolina_functions.py

```diff
@@ -97,35 +97,31 @@ def shortest_paths(track_graph : Graph) -> Dict[Voxel, Dict[Voxel, float]]:
     """Compute shortest path lengths between all nodes in a weighted graph."""
     return nx.all_pairs_dijkstra_path_length(track_graph, weight='distance')
 
-
-def find_extrema(distance : Dict[Voxel, Dict[Voxel, float]]) -> Tuple[Voxel, Voxel]:
-    """Find the extrema of the track."""
+def find_extrema_and_length(distance : Dict[Voxel, Dict[Voxel, float]]) -> Tuple[Voxel, Voxel, float]:
+    """Find the extrema and the length of a track, given its dictionary of distances."""
     if not distance:
         raise NoVoxels
     if len(distance) == 1:
         only_voxel = next(iter(distance))
-        return (only_voxel, only_voxel)
+        return (only_voxel, only_voxel, 0.)
     first, last, max_distance = None, None, 0
     for source, target in combinations(distance, 2):
         d = distance[source][target]
         if d > max_distance:
             first, last, max_distance = source, target, d
-    return (first, last)
+    return (first, last, max_distance)
 
+def find_extrema(track: Graph) -> Tuple[Voxel, Voxel]:
+    """Find the extrema of a track."""
+    distances = shortest_paths(track)
+    extrema_and_length = find_extrema_and_length(distances)
+    return (extrema_and_length[0], extrema_and_length[1])
 
-def length(distance : Dict[Voxel, Dict[Voxel, float]]) -> float:
+def length(track: Graph) -> float:
     """Calculate the length of a track."""
-    if not distance:
-        raise NoVoxels
-    if len(distance) == 1:
-        return 0.
-    max_distance = 0.
-    for source, target in combinations(distance, 2):
-        d = distance[source][target]
-        if d > max_distance:
-            max_distance = d
-
-    return  max_distance
+    distances = shortest_paths(track)
+    extrema_and_length = find_extrema_and_length(distances)
+    return extrema_and_length[2]
```

Comment:
The *magic number* `2`is really unhelpful. I would prefer something like

    _, _, length = find_extrema_and_length(distances)
    return length